mkosi/__init__.py

```diff
@@ -4434,10 +4514,13 @@ def load_args(args: CommandLineArguments) -> CommandLineArguments:
         elif args.distribution == Distribution.openmandriva:
             args.release = "cooker"
 
-    if (args.distribution in (Distribution.centos, Distribution.centos_epel) and
-        args.release == "8" and
-        args.output_format == OutputFormat.gpt_btrfs):
-        die("Sorry, CentOS 8 does not support btrfs")
+    if args.distribution in (Distribution.centos, Distribution.centos_epel):
+        epel_release = int(args.release.split('.')[0])
+        if epel_release <= 8 and args.output_format == OutputFormat.gpt_btrfs:
+            die(f"Sorry, CentOS {epel_release} does not support btrfs")
+        if epel_release <= 7 and args.bootable and 'uefi' in args.boot_protocols and args.with_unified_kernel_images:
+            warn(f"Using --without-unified-kernel-images as CentOS {epel_release} does not support unified kernel images.")
+            args.with_unified_kernel_images = False
```

Comment:
I'd prefer simply failing early if `--without-unified-kernel-images` is not specified for either the bootloader option or centos 7 (depending on if you're up for adding the bootloader option).

I know I added the Clear Linux automatic fallback but I already removed that (or its in one of the open PRs) that. It adds to much magic and I'd prefer users specifically acknowledging and being okay with centos 7 not doing unified kernel images rather than automatically assume they will be ok with it just because centos 7 happens to not support them.

Reply:
Fine by me, I'd actually prefer that too.